src/components/product.js

```diff
@@ -52,6 +52,22 @@ function whitelistedProperties(selectorStyles) {
   }, {});
 }
 
+function adaptConfig(config) {
+  if (config.id && !isBase64(config.id)) {
+    config.id = btoa(`gid://shopify/Product/${config.id}`);
+  }
+
+  return config;
+}
+
+function isBase64(str) {
+  try {
+    return btoa(atob(str)) === str;
```

Comment:
i like this `storefrontId` idea. definitely way nicer than trying to detect what type of ID was provided.